### PR TITLE
EstimApi support

### PR DESCRIPTION
**Pull request by xman2B**
Integrate EstimApi into TeaseAI Java.
This adds a new JS function estimApi, which can be used by personalities and an estim metronome, which can be configured in the settings.

### CODE REVIEW

File: src/me/goddragon/teaseai/api/session/Session.java [hunk 1]
```diff
@@ -24,6 +41,21 @@
     private boolean haltSession = false;
     private long startedAt;
 
+    private EstimAPI estimAPI = null;
+    private List<Mode> estimEnabledModes = null;
+    private Map<Channel, Integer> estimChannelValues = new HashMap<Channel, Integer>();
+    
+    private List<Mode> string2List(String string) {
```
Comment (GodDragoner) [inline src/me/goddragon/teaseai/api/session/Session.java]: Own class for this? Util class? Estim Class?

File: src/me/goddragon/teaseai/api/session/StrokeHandler.java [hunk 2]
```diff
@@ -42,6 +45,10 @@ public void startMetronome(int bpm, int durationSeconds) {
         currentBPM = bpm;
 
         (metronome = new Metronome()).start(bpm);
+        
+        if (TeaseAI.application.ESTIM_ENABLED.getBoolean() && TeaseAI.application.ESTIM_METRONOME.getBoolean()) {
```
Comment (GodDragoner) [inline src/me/goddragon/teaseai/api/session/StrokeHandler.java]: Wait there is an EstimAPI instance in the Session already. Why not use that one?

**Review by GodDragoner — COMMENTED** [review]
Please change those things/give a reason why it should be this way

File: src/me/goddragon/teaseai/api/session/Session.java (hunk 1 above)
Comment (xman2B) [reply src/me/goddragon/teaseai/api/session/Session.java]: Done. It is now in EstimState.

File: src/me/goddragon/teaseai/api/session/StrokeHandler.java (hunk 2 above)
Comment (xman2B) [reply src/me/goddragon/teaseai/api/session/StrokeHandler.java]: It wasn't the same class. Renamed it to EstimMetronome to avoid confusion.

**Comment by skier233** [comment]
You didn't include the jar file resource in the Resources folder. So it currently won't build. @xman2B  @GodDragoner

**Comment by skier233** [comment]
> You didn't include the jar file resource in the Resources folder. So it currently won't build. @xman2B @GodDragoner

I went ahead and fixed it. Just watch out in the future.

**Pull request merged**